src/main/java/uk/gov/hmcts/reform/hmc/model/CaseDetails.java

```diff
@@ -0,0 +1,66 @@
+package uk.gov.hmcts.reform.hmc.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.hibernate.validator.constraints.URL;
+import uk.gov.hmcts.reform.hmc.exceptions.ValidationError;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
+
+@Data
+@NoArgsConstructor
+public class CaseDetails {
+
+    @NotEmpty(message = ValidationError.HMCTS_SERVICE_CODE_EMPTY_INVALID)
+    @Pattern(regexp = "^\\w{4}$", message = ValidationError.HMCTS_SERVICE_CODE_EMPTY_INVALID)
+    private String hmctsServiceCode;
+
+    @NotEmpty(message = ValidationError.CASE_REF_EMPTY)
+    @Pattern(regexp = "^\\d{16}$", message = ValidationError.CASE_REF_INVALID)
+    private String caseRef;
+
+    @NotEmpty(message = ValidationError.REQUEST_TIMESTAMP_EMPTY)
+    private String requestTimeStamp;
```

Comment:
```suggestion
    private LocalDateTime requestTimeStamp;
```
We are using `LocalDateTime` in one place - would make sense to use it everywhere? I've added a couple of comments to places where it seems `String` is used, but not all.